src/simulator/halo_simulator.py

```diff
@@ -213,6 +214,46 @@ def simulated_reach_by_spend(
         kplus_reaches = ReachPoint.frequencies_to_kplus_reaches(frequencies)
         return ReachPoint(impressions, kplus_reaches, spends)
 
+    def _liquid_legions_cardinality_estimate_variance(self, n: int) -> float:
+        """Variance of cardinality estimate by unnoised LiquidLegions.
+
+        Args:
+          n:  Carinality of the items contained in a LiquidLegions.
+
+        Returns:
+          var(n-hat) where n-hat is the cardinality estimate from a
+          LiquidLegions which contains n distinct items and has the system
+          parameters (decay rate and sketch size) in this halo.
+        """
+        m = self._params.liquid_legions.sketch_size
+        a = self._params.liquid_legions.decay_rate
+        c = a * n / m / (1 - np.exp(-a))
+        variance = (
+            expi(-c) - expi(-c * np.exp(-a)) - expi(-2 * c) + expi(-2 * c * np.exp(-a))
+        )
+        variance *= a * n ** 2 / m / (np.exp(-c) - np.exp(-c * np.exp(-a))) ** 2
+        variance -= n
+        return variance
+
+    def _liquid_legions_num_active_regions(
+        self, n: int, random_generator: np.random.Generator = np.random.default_rng()
```

Comment:
I know we have done this several times, but I always wonder why we need to have a different random generator when there is one in `self._param`.

Reply:
Thank you for pointing this out!  Revised following your practice in PR 56.